src/encoder.rs

```diff
@@ -357,10 +356,10 @@ impl<'a> Encoder<'a> {
     let creation: u32 = FromPyObject::extract(self.py, &py_creation)?;
 
     self.data.push(consts::TAG_NEW_PID_EXT);
-    self.write_atom_from_cow(node_name.to_string(self.py)?).unwrap();
-    self.data.write_u32::<BigEndian>(id).unwrap();
-    self.data.write_u32::<BigEndian>(serial).unwrap();
-    self.data.write_u32::<BigEndian>(creation).unwrap();
+    self.write_atom_from_cow(node_name.to_string(self.py)?).map_err(|e| CodecError::IOError{txt: format!("{:?}", e)})?;
```

Comment:
if all these are maps from std::io::Error into CodecError, just implement From for it, once.